middleware/markdown/page.go

```diff
@@ -0,0 +1,161 @@
+package markdown
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/russross/blackfriday"
+)
+
+const (
+	// Date format YYYY-MM-DD HH:MM:SS
+	timeLayout = `2006-01-02 15:04:05`
+
+	// Length of page summary.
+	summaryLen = 150
+)
+
+// PageLink represents a statically generated markdown page.
+type PageLink struct {
+	Title   string
+	Summary string
+	Date    time.Time
+	Url     string
+}
+
+// pageLinkSorter sorts PageLink by newest date to oldest.
+type pageLinkSorter []PageLink
+
+func (p pageLinkSorter) Len() int           { return len(p) }
+func (p pageLinkSorter) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+func (p pageLinkSorter) Less(i, j int) bool { return p[i].Date.After(p[j].Date) }
```

Comment:
Could we rename this type to something that indicates it sorts by Date? Kind of like the first example here: http://golang.org/pkg/sort/